### PR TITLE
Feature/moov 2687 pre authorisation screen

### PR DESCRIPTION
**Pull request by Publicker**
This PR introduces a new Pre-Authorisation screen for Payruns, focusing on the UI aspect to enhance the payrun process flow within the application. This screen provides users with detailed information on the Payrun and includes options to proceed with the authorisation process.

# Changes
- **New Pre-Authorisation Screen**: Implemented the UI for the Pre-Authorisation screen accessible through the "Send request" flow in the Payrun Details screen (this was done like this just for testing reasons, after we have the email mechanism where user will get this URL this will get removed).

# Important Notes
- **Non-Functional State Changes**: Any modifications made in the Payrun Details screen are currently not integrated to affect the Payrun data presented in the Pre-Authorisation screen. This functionality will be developed and integrated in future updates.
- **Selected Accounts Placeholder**: The selected accounts section will appear empty for now. This is in anticipation of future integration where changes from the Payrun Details will reflect in this screen.
- **Inactive Buttons**: The "Reject" and "Accept and continue" buttons are present for UI completeness but are non-functional pending backend integration.

# Demo Video

https://github.com/nofrixion/nofrixion.business/assets/52673485/6e2b8195-3593-47e3-8f2c-c2344509d910

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/196.

File: apps/business/src/pages/authLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      setMerchants(merchantsResponse.data)
+
+      if (!merchant || !merchants?.find((m) => m.id === merchant.id)) {
+        if (merchants && merchants.length > 0) {
+          setMerchant(merchants[0])
+        }
+      }
+    } else if (merchantsResponse?.status === 'error') {
+      console.log('Error fetching merchants', merchantsResponse.error)
+      authState?.logOut && authState.logOut()
```
Comment (donalnofrixion) [inline apps/business/src/pages/authLayout.tsx:44]: I don't think we're doing this anymore. We display an error dialog instead

File: apps/business/src/pages/authLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { useAuth } from '../lib/auth/useAuth'
+import useMerchantsStore from '../lib/stores/useMerchantsStore'
+import useMerchantStore from '../lib/stores/useMerchantStore'
+import useUserStore from '../lib/stores/useUserStore'
+
+interface AuthLayoutProps {
+  children: React.ReactNode
+}
+
+const AuthLayout: React.FC<AuthLayoutProps> = ({ children }) => {
```
Comment (donalnofrixion) [inline apps/business/src/pages/authLayout.tsx:15]: Can it be called something else? `AuthLayout` makes me think it's to do with authorisation

**Review by donalnofrixion — CHANGES_REQUESTED** [review]

File: apps/business/src/pages/authLayout.tsx (hunk 1 above)
Comment (Publicker) [reply apps/business/src/pages/authLayout.tsx:44]: Maybe `SessionAuthProvider`?

File: apps/business/src/pages/authLayout.tsx (hunk 2 above)
Comment (Publicker) [reply apps/business/src/pages/authLayout.tsx:15]: Maybe SessionAuthProvider?

File: apps/business/src/pages/authLayout.tsx (hunk 1 above)
Comment (Publicker) [reply apps/business/src/pages/authLayout.tsx:44]: Alright - I will get rid of it

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/196.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/196.

File: apps/business/src/pages/authLayout.tsx (hunk 1 above)
Comment (Publicker) [reply apps/business/src/pages/authLayout.tsx]: Created ticket for this and also discussing with @pabloi-nf what to do in this case. Exploring the option of showing a modal

**Review by donalnofrixion — APPROVED** [review]

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Thanks Pablo, it looks very good.

A few comments:

- The color of the status chip should be different. This is because we are using it on top of gray background instead of white, so we need to make the colours more vibrant.
> image
> image

- The font size of the "Notes for authoriser" needs to be 13px.
- Let's change the font-weight of the lorem ipsum text to 400. This was a mistake on the Figma.
- Make sure to align the content in the same vertical line. 
> 
image

The text on top is currently aligned to the box, not to the content inside the box. 

For rounded buttons, the margin should be just 1rem from the edge of the box
> image


This is the guide https://www.figma.com/file/Sc96Upmn3MpOmvM9jgvaUG/MMB-Portal-%2F-HiFi?type=design&node-id=11549%3A57306&mode=dev&t=SAT6zUEzpcsLfBbZ-1

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/196.

**Review by arifmatin — APPROVED** [review]

**Comment by Publicker** [comment]
Thanks @pabloi-nf, should be good now.

**Review by pabloi-nf — APPROVED** [review]

**Pull request merged**